MapboxNavigation/RouteMapViewController.swift

```diff
@@ -258,6 +260,34 @@ extension RouteMapViewController: NavigationMapViewDelegate {
             return defaultReturn
         }
         
+        if let userLocation = mapView.userLocation, routeController.showCurrentWayNameLabel == true {
+            
+            // Somehow check the current style has mapbox streets
+            //
+            //let streets = MGLSource(identifier: "mapbox://mapbox-streets-v7")
+            //guard mapView.style!.sources.contains(streets) else {
+            //  return
+            //}
+            
+            let userPuck = mapView.view(for: userLocation)
+            if let userPuck = userPuck {
+                let features = mapView.visibleFeatures(in: userPuck.frame)
+                
+                for layer in features {
+                    
+                    // TODO: check for acceptable classes
+                    // if let feature = layer as? MGLPolylineFeature,
```

Comment:
MGLPolylineFeature is a kind of feature; a feature is part of a shape or vector source. But note for example that a single vector source might include features of many types at the same time.

You want to find out which style layers would display roads, which are linear. A style layer that renders linear shapes is an MGLLineStyleLayer. After you filter the style’s list of layers to include only MGLLineStyleLayers, you can further filter the list to layers whose source is Mapbox Streets and whose source layer identifier is [`road`](https://www.mapbox.com/vector-tiles/mapbox-streets-v7/#road).

You could also filter the layers by their predicates, but note that predicates can be arbitrarily complex, so this might be more trouble than it’s worth. For example, if you wanted to know whether a layer’s predicate includes only motorways, you’d have to account for predicates such as `(class IN {'motorway_link', 'trunk_link'} AND class != 'motorway' AND oneway = FALSE) OR layer = -2` etc.